src/Engine/Engine.h

```diff
@@ -77,10 +78,10 @@ class Engine {
     std::vector<std::string>                 mattrs;
     VMInstanceRef                            vminstance;
 
-    Assembly*                                                       assembly;
-    ExecBlockManager*                                               blockManager;
-    ExecBroker*                                                     execBroker;
-    std::vector<std::shared_ptr<PatchRule>>                         patchRules;
+    std::unique_ptr<Assembly>                                       assembly;
```

Comment:
Do we really need pointer ?

Reply:
The used of pointer allows to destruct assembly and recreate it. It's needed to perform the copy of the VM if the copied VM doesn't have the same llvm parameters (``cpu`` and ``mattrs``).